cmd/kube-controller-manager/app/core.go

```diff
@@ -81,21 +82,39 @@ func startServiceController(ctx ControllerContext) (http.Handler, bool, error) {
 }
 
 func startNodeIpamController(ctx ControllerContext) (http.Handler, bool, error) {
-	var clusterCIDR *net.IPNet
 	var serviceCIDR *net.IPNet
+	var err error
+	clusterCIDRs := make([]*net.IPNet, 0)
 
 	if !ctx.ComponentConfig.KubeCloudShared.AllocateNodeCIDRs {
 		return nil, false, nil
 	}
 
-	var err error
+	// cluster cidr list processing
+	//TODO (khenidak) clustercidr processing is the same
```

Comment:
See above, maybe we could put this function into a new pkg/util/network/network.go or something like that.